packages/host/app/services/matrix-service.ts

```diff
@@ -407,6 +407,55 @@ export default class MatrixService extends Service {
     return users;
   }
 
+  // the matrix SDK is using an old version of this API that 
+  // doesn't provide login via email, so we use the API directly
+  async loginWithEmail(
+    email: string,
+    password: string,
+  ) {
+    let response = await fetch(
+      `${matrixURL}/_matrix/client/v3/login`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          identifier: {
+            type: "m.id.thirdparty",
+            medium: "email",
+            address: email,
+          },
+          password,
+          type: "m.login.password",
+        }),
+      },
+    );
+    if (response.ok) {
+      return (await response.json()) as MatrixSDK.IAuthData;
+    } else {
+      let data = (await response.json()) as { errcode: string; error: string };
+      let error = new Error(data.error) as any;
+      error.data = data;
+      error.status = response.status;
+      throw error;
+    }
+  }
+
+  async login(usernameOrEmail: string, password: string) {
+    try {
+      const cred = await this.client.loginWithPassword(usernameOrEmail, password);
```

Comment:
Is this supposed to be `loginWithUsername`?

Reply:
that function is from matrix client

Reply:
I see... it looked confusing to me to have `loginWithPassword` and `loginWithEmail` when password is actually needed in both cases

Reply:
so the idea is that `loginWithPassword()` uses a password for authorization. loginWithEmail() uses email for authorization: a magic link sent to your email that when clicked on let's you login without a password

Reply:
👍